From 21f3c1bfb946eb56599a9dcfc2c5bf4b4b1649fe Mon Sep 17 00:00:00 2001
From: David Sniatkiwicz <davidsn@marvell.com>
Date: Tue, 5 Sep 2017 10:33:59 +0300
Subject: [PATCH 1238/1345] dts: arm64: marvell: update device tree files for
 the armada 7K/8K dual cpu

commit  f6edf3384d4e93066b06d8016601602a7c9211e9 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

align armada 7K/8K dual cpu setups with armada 7K/8K quad cpu setups.

Change-Id: Idb0e4d5e0185f4344490c88fcd1053ab5e53b7dd
Signed-off-by: David Sniatkiwicz <davidsn@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/43762
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
(cherry picked from commit 26ff0a4c5f4b10024d14db30a4a1df80adf87536)
Reviewed-on: http://vgitil04.il.marvell.com:8080/44333
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-7020-amc.dts    |   25 ---
 arch/arm64/boot/dts/marvell/armada-7020.dtsi       |   82 ++++++++++
 arch/arm64/boot/dts/marvell/armada-8020.dtsi       |  152 ++++++++++++++++++
 arch/arm64/boot/dts/marvell/armada-ap806-dual.dtsi |  167 ++++++++++++++++++++
 4 files changed, 401 insertions(+), 25 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7020-amc.dts b/arch/arm64/boot/dts/marvell/armada-7020-amc.dts
index 1776545..9aef7b0b 100644
--- a/arch/arm64/boot/dts/marvell/armada-7020-amc.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7020-amc.dts
@@ -57,14 +57,6 @@
 
 	chosen { };
 
-	aliases {
-		gpio1 = &cpm_gpio0;
-		gpio2 = &cpm_gpio1;
-		ethernet0 = &emac0;
-		ethernet1 = &emac2;
-		ethernet2 = &emac3;
-	};
-
 	ap806 {
 		config-space {
 			spi0: spi@510600 {
@@ -85,27 +77,11 @@
 	cpn-110-master {
 		config-space {
 			pinctrl@440000 {
-				compatible = "marvell,a70x0-pinctrl";
-
-				i2c0_pins: i2c-pins-0 {
-					marvell,pins = "mpp37", "mpp38";
-					marvell,function = "i2c0";
-				};
-
 				i2c1_pins: i2c-pins-1 {
 					marvell,pins = "mpp50", "mpp51";
 					marvell,function = "i2c1";
 				};
 
-				cpm_sdhci_pins: cpm-sdhi-pins {
-					marvell,pins = "mpp49", "mpp56", "mpp57", "mpp58",
-							"mpp59", "mpp60", "mpp61";
-					marvell,function = "sdio";
-				};
-			};
-
-			rtc@284000 {
-				status = "okay";
 			};
 
 			usb3h0: usb3@500000 {
@@ -118,7 +94,6 @@
 
 			sata: sata@540000 {
 				status = "okay";
-				#stream-id-cells = <1>;
 
 				sata-port@1 {
 					phys = <&comphy0 5 COMPHY_SATA1>;
diff --git a/arch/arm64/boot/dts/marvell/armada-7020.dtsi b/arch/arm64/boot/dts/marvell/armada-7020.dtsi
index f2d127e..0403f50 100644
--- a/arch/arm64/boot/dts/marvell/armada-7020.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7020.dtsi
@@ -53,7 +53,89 @@
 	compatible = "marvell,armada7020", "marvell,armada-ap806-dual",
 		     "marvell,armada-ap806";
 
+	aliases {
+		gpio1 = &cpm_gpio0;
+		gpio2 = &cpm_gpio1;
+		ethernet0 = &emac0;
+		ethernet1 = &emac2;
+		ethernet2 = &emac3;
+	};
+
+	ap806 {
+		config-space {
+			smmu: iommu@5000000 {
+				mmu-masters = <&usb3h0 0x483>,
+					      <&usb3h1 0x484>,
+					      <&sata 0x485>,
+					      <&cp_xor0 0x489>,
+					      <&cp_xor1 0x48a>;
+			};
+			pinctrl@6f4000 {
+				ap_emmc_pins: emmc-pins-0 {
+					marvell,pins = "mpp0", "mpp1", "mpp2",
+					"mpp3", "mpp4", "mpp5";
+					marvell,function = "sdio";
+				};
+			};
+		};
+	};
+
 	cp110-master {
+		config-space {
+			pinctrl@440000 {
+			compatible = "marvell,a70x0-pinctrl";
+				cpm_tdm_pins: tdm-pins {
+					marvell,pins = "mpp0", "mpp1", "mpp2",
+					"mpp3", "mpp4", "mpp5";
+					marvell,function = "tdm";
+				};
+				i2c0_pins: i2c-pins-0 {
+					marvell,pins = "mpp37", "mpp38";
+					marvell,function = "i2c0";
+				};
+				cpm_sdhci_pins: cpm-sdhi-pins {
+					marvell,pins = "mpp56", "mpp57", "mpp58",
+						       "mpp59", "mpp60", "mpp61";
+					marvell,function = "sdio";
+				};
+				/* following VBUS IN/OUT GPIO entries is not used
+				 * (for reference only) GPIO is not connected
+				 * in default A7040-DB board
+				 */
+				xhci0_vbus_pins: xhci0-vbus-pins {
+					marvell,pins = "mpp12";
+					marvell,function = "gpio";
+				};
+			};
+			rtc@284000 {
+				status = "okay";
+			};
+			ppv22@000000 {
+				dma-coherent;
+			};
+			sata: sata@540000 {
+				#stream-id-cells = <1>;
+			};
+			usb3h0: usb3@500000 {
+				#stream-id-cells = <1>;
+			};
+			usb3h1: usb3@510000 {
+				#stream-id-cells = <1>;
+			};
+			cp_xor0: cpxor0@6a0000 {
+				#stream-id-cells = <1>;
+			};
+			cp_xor1: cpxor1@6c0000 {
+				#stream-id-cells = <1>;
+			};
+			/* AXI bus of IHB1 is not connected in Armada-7040 package. */
+			axim-hb1-rd@3c8000 {
+				status = "disabled";
+			};
+			axim-hb1-wr@3c9000 {
+				status = "disabled";
+			};
+		};
 		pcie@0x600000 {
 			msi-parent = <&gic_v2m0>;
 		};
diff --git a/arch/arm64/boot/dts/marvell/armada-8020.dtsi b/arch/arm64/boot/dts/marvell/armada-8020.dtsi
index 7c08f1f..deeecdf 100644
--- a/arch/arm64/boot/dts/marvell/armada-8020.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8020.dtsi
@@ -53,6 +53,158 @@
 	model = "Marvell Armada 8020";
 	compatible = "marvell,armada8020", "marvell,armada-ap806-dual",
 		     "marvell,armada-ap806";
+
+	aliases {
+		gpio0 = &cpm_gpio0;
+		gpio1 = &cpm_gpio1;
+		ethernet0 = &emac0;
+		ethernet1 = &emac2;
+		ethernet2 = &emac3;
+		ethernet3 = &emac0_1;
+		ethernet4 = &emac2_1;
+		ethernet5 = &emac3_1;
+	};
+
+	ap806 {
+		config-space@f0000000 {
+			/* Temp: disable all Coresight nodes */
+			etm0@c40000 {
+				status = "disabled";
+			};
+			etm1@d40000 {
+				status = "disabled";
+			};
+			etf0@809000 {
+				status = "disabled";
+			};
+			etf1@80a000 {
+				status = "disabled";
+			};
+			funnel@810000 {
+				status = "disabled";
+			};
+			etf5@811000 {
+				status = "disabled";
+			};
+			etr@812000 {
+				status = "disabled";
+			};
+			smmu: iommu@5000000 {
+				mmu-masters = <&usb3h0 0x483>,
+					      <&usb3h1 0x484>;
+			};
+			pinctrl@6f4000 {
+				ap_emmc_pins: emmc-pins-0 {
+					marvell,pins = "mpp0", "mpp1", "mpp2",
+					"mpp3", "mpp4", "mpp5", "mpp6", "mpp7",
+					"mpp8", "mpp9", "mpp10";
+					marvell,function = "sdio";
+				};
+				ap_spi0_pins: spi-pins-0 {
+					marvell,pins = "mpp0", "mpp1", "mpp2",
+					"mpp3";
+					marvell,function = "spi0";
+				};
+			};
+		};
+	};
+
+	cp110-master {
+		config-space@f2000000 {
+			ppv22@000000 {
+				dma-coherent;
+			};
+			pinctrl@440000 {
+				compatible = "marvell,a80x0-cp0-pinctrl";
+
+				cpm_i2c0_pins: i2c-pins-0 {
+					marvell,pins = "mpp37", "mpp38";
+					marvell,function = "i2c0";
+				};
+				cpm_i2c1_pins: i2c-pins-1 {
+					marvell,pins = "mpp35", "mpp36";
+					marvell,function = "i2c1";
+				};
+				cpm_ge2_rgmii_pins: ge-rgmii-pins-0 {
+					marvell,pins = "mpp44", "mpp45", "mpp46",
+						       "mpp47", "mpp48", "mpp49",
+						       "mpp50", "mpp51", "mpp52",
+						       "mpp53", "mpp54", "mpp55";
+					marvell,function = "ge1";
+				};
+				pca0_pins: pca0_pins {
+					marvell,pins = "mpp62";
+					marvell,function = "gpio";
+				};
+				cpm_sdhci_pins: cpm-sdhi-pins {
+					marvell,pins = "mpp56", "mpp57", "mpp58",
+						       "mpp59", "mpp60", "mpp61";
+					marvell,function = "sdio";
+				};
+			};
+			usb3h0: usb3@500000 {
+				#stream-id-cells = <1>;
+			};
+			usb3h1: usb3@510000 {
+				#stream-id-cells = <1>;
+			};
+		};
+		pcie@0x600000 {
+			msi-parent = <&gic_v2m0>;
+		};
+		pcie@0x620000 {
+			msi-parent = <&gic_v2m0>;
+		};
+		pcie@0x640000 {
+			msi-parent = <&gic_v2m0>;
+		};
+	};
+
+	cp110-slave {
+		config-space@f4000000 {
+			ppv22@000000 {
+				dma-coherent;
+			};
+			pinctrl@440000 {
+				compatible = "marvell,a80x0-cp1-pinctrl";
+
+				cps_ge1_rgmii_pins: ge-rgmii-pins-0 {
+					marvell,pins = "mpp0", "mpp1", "mpp2",
+						       "mpp3", "mpp4", "mpp5",
+						       "mpp6", "mpp7", "mpp8",
+						       "mpp9", "mpp10", "mpp11";
+					marvell,function = "ge0";
+				};
+				cps_tdm_pins: tdm-pins {
+					marvell,pins = "mpp0", "mpp1", "mpp2",
+						       "mpp3", "mpp4", "mpp5";
+					marvell,function = "tdm";
+				};
+				cps_spi0_pins: spi-pins-0 {
+					marvell,pins = "mpp7", "mpp8", "mpp9",
+						       "mpp10", "mpp11";
+					marvell,function = "spi0";
+				};
+				cps_spi1_pins: spi-pins-1 {
+					marvell,pins = "mpp13", "mpp14", "mpp15",
+						       "mpp16";
+					marvell,function = "spi1";
+				};
+			};
+			rtc@284000 {
+				status = "okay";
+			};
+		};
+		pcie@0x600000 {
+			msi-parent = <&gic_v2m0>;
+		};
+		pcie@0x620000 {
+			msi-parent = <&gic_v2m0>;
+		};
+		pcie@0x640000 {
+			msi-parent = <&gic_v2m0>;
+		};
+	};
 };
 
 /* The RTC requires external oscillator. But on Aramda 80x0, the RTC clock
diff --git a/arch/arm64/boot/dts/marvell/armada-ap806-dual.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806-dual.dtsi
index b7694b7..fb1346f 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806-dual.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806-dual.dtsi
@@ -92,4 +92,171 @@
 
 		};
 	};
+
+	ap806 {
+		config-space@f0000000 {
+			etm0@c40000 {
+				compatible = "arm,coresight-etm4x", "arm,primecell";
+				reg = <0xc40000 0x1000>;
+
+				cpu = <&cpu0>;
+				clocks = <&syscon 3>;
+				clock-names = "apb_pclk";
+				port {
+					etm0_out: endpoint {
+						remote-endpoint = <&etf0_in>;
+					};
+				};
+			};
+
+			etm1@d40000 {
+				compatible = "arm,coresight-etm4x", "arm,primecell";
+				reg = <0xd40000 0x1000>;
+
+				cpu = <&cpu1>;
+				clocks = <&syscon 3>;
+				clock-names = "apb_pclk";
+				port {
+					etm1_out: endpoint {
+						remote-endpoint = <&etf1_in>;
+					};
+				};
+			};
+
+
+			etf0@809000 {
+				compatible = "arm,coresight-tmc", "arm,primecell";
+				reg = <0x809000 0x1000>;
+				clocks = <&syscon 3>;
+				clock-names = "apb_pclk";
+
+				ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+
+					port@0 {
+						reg = <0>;
+						etf0_in: endpoint {
+							slave-mode;
+							remote-endpoint = <&etm0_out>;
+						};
+					};
+
+					port@1 {
+						reg = <0>;
+						etf0_out: endpoint {
+							remote-endpoint = <&funnel_in_0>;
+						};
+					};
+				};
+			};
+
+			etf1@80a000 {
+				compatible = "arm,coresight-tmc", "arm,primecell";
+				reg = <0x80a000 0x1000>;
+				clocks = <&syscon 3>;
+				clock-names = "apb_pclk";
+
+				ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+
+					port@0 {
+						reg = <0>;
+						etf1_in: endpoint {
+							slave-mode;
+							remote-endpoint = <&etm1_out>;
+						};
+					};
+
+					port@1 {
+						reg = <0>;
+						etf1_out: endpoint {
+							remote-endpoint = <&funnel_in_1>;
+						};
+					};
+				};
+			};
+
+			funnel@810000 {
+				compatible = "arm,coresight-funnel", "arm,primecell";
+				reg = <0x810000 0x1000>;
+				clocks = <&syscon 3>;
+				clock-names = "apb_pclk";
+
+				ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+
+					/* funnel output port */
+					port@0 {
+						reg = <0>;
+						funnel_out: endpoint {
+							remote-endpoint = <&etf5_in>;
+						};
+					};
+
+					/* funnel input ports */
+					port@1 {
+						reg = <0>;
+						funnel_in_0: endpoint {
+							slave-mode;
+							remote-endpoint = <&etf0_out>;
+						};
+					};
+
+					port@2 {
+						reg = <1>;
+						funnel_in_1: endpoint {
+							slave-mode;
+							remote-endpoint = <&etf1_out>;
+						};
+					};
+
+				};
+			};
+
+			etf5@811000 {
+				compatible = "arm,coresight-tmc", "arm,primecell";
+				reg = <0x811000 0x1000>;
+				clocks = <&syscon 3>;
+				clock-names = "apb_pclk";
+
+				ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+
+					port@0 {
+						reg = <0>;
+						etf5_in: endpoint {
+							slave-mode;
+							remote-endpoint = <&funnel_out>;
+						};
+					};
+
+					port@1 {
+						reg = <0>;
+						etf5_out: endpoint {
+							remote-endpoint = <&etr_in>;
+						};
+					};
+				};
+			};
+
+
+			etr@812000 {
+				compatible = "arm,coresight-tmc", "arm,primecell";
+				reg = <0x812000 0x1000>;
+				clocks = <&syscon 3>;
+				clock-names = "apb_pclk";
+
+				port {
+					etr_in: endpoint {
+						slave-mode;
+						remote-endpoint = <&etf5_out>;
+					};
+				};
+			};
+		};
+	};
 };
-- 
1.7.9.5

